src/Services/CandidateService.php

```diff
@@ -0,0 +1,191 @@
+        $this->objectFactory = $objectFactory;
+    }
+
+    /**
+     * @param int $id
+     * @return bool
+     */
+    public function startQuiz(int $id): bool
+    {
+        $quizTemplate = $this->quizTemplateService->getQuiz($id);
+
+        $quizInstance = $this->objectFactory->createEntityFromObject($quizTemplate);
+        if (!$quizInstance->save()) {
+            return false;
+        }
+
+        foreach ($quizTemplate->getQuestions() as $question) {
+            $questionInstance = $this->objectFactory->createEntityFromObject($question);
+            $questionInstance->setQuizInstanceId($quizInstance->getId());
+            if (!$questionInstance->save()) {
+                return false;
+            }
+
+            if (!$this->createAnswer($question->getAnswerText(), $questionInstance->getId())) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * @param int $quizTemplateId
+     * @param int $userId
+     * @return bool
+     */
+    public function getQuizInstanceInProgress(int $quizTemplateId, int $userId): ?QuizInstance
+    {
+        return $this->findQuizInProgress($quizTemplateId, $userId);
+    }
+
+    /**
+     * @param int $quizId
+     * @param int $question
+     * @return QuestionInstanceDTO
+     */
+    public function getQuestion(int $quizId, int $question): QuestionInstanceDTO
+    {
+        $questionRepo = $this->repoManager->getRepository(QuestionInstance::class);
+        $answerRepo = $this->repoManager->getRepository(TextInstance::class);
+        $question = $questionRepo->findBy
+        (
+            [
+                'quizInstanceId' => $quizId
+            ],
+            [],
+            $question - 1,
+            1
+        );
+        $answerTemplate = $answerRepo->findBy(['questionInstanceId' => $question[0]->getId()], [], 0, 1);
+
+        return QuestionInstanceDTO::createWithObject($question[0], $answerTemplate[0]);
+    }
+
+    /**
+     * @param int $quizId
+     * @return QuizInstanceDTO
+     */
+    public function getQuiz(int $quizId): QuizInstanceDTO
+    {
+        $quizRepo = $this->repoManager->getRepository(QuizInstance::class);
+        $quizInstance = $quizRepo->find($quizId);
+
+        return QuizInstanceDTO::createWithObject($quizInstance, $this->findQuestionsForQuiz($quizId));
+    }
+
+    /**
+     * This function will return the number of first question of quiz which is not answered
+     *
+     * @param int $quizTemplateId
+     * @param int $userId
+     * @return int|null
+     */
+    public function getFirstUnansweredQuestionId(QuizInstance $quizInstanceInProgress): ?int
+    {
+        $questionRepo = $this->repoManager->getRepository(QuestionInstance::class);
+        $question = $questionRepo->findOneBy
+        (
+            [
+                'quizInstanceId' => $quizInstanceInProgress->getId(),
+                'isAnswered' => 0
+            ]
+        );
+        if(!$question->getId()){
```

Comment:
```suggestion
        return $question->getId();
```

I expect it will be null if it's not set?